### PR TITLE
Feature/pom xml support

### PR DESCRIPTION
**Pull request by apirogov**
Implementation for pom.xml

Includes fancy wrapper for dict-like access to XML to allow tomlkit-like updating features using python builtin xml elementtree.

Also some refactorings reducing copy-paste code

### CODE REVIEW

File: src/somesy/commands/sync.py [hunk 1]
```diff
@@ -1,153 +1,63 @@
...
 
-    # create these by default if they are missing:
-    if not conf.no_sync_cff:
-        _sync_cff(metadata, conf.cff_file)
+    if conf.julia_file.is_file() and not conf.no_sync_julia:
+        _sync_file(metadata, conf.julia_file, Julia)
 
-    if not conf.no_sync_codemeta:
-        _sync_codemeta(metadata, conf.codemeta_file)
+    if not conf.no_sync_fortran:
```
Comment (mustafasoylu) [inline src/somesy/commands/sync.py:51]: shouldnt be this?
```python
if conf.fortran_file.is_file() and not conf.no_sync_fortran:
```

File: src/somesy/core/writer.py [hunk 2]
```diff
@@ -249,6 +264,9 @@ def sync(self, metadata: ProjectMetadata) -> None:
         self.maintainers = self._sync_person_list(
             self.maintainers, metadata.maintainers()
         )
+        # self.contributors = self._sync_person_list(
```
Comment (mustafasoylu) [inline src/somesy/core/writer.py:267]: should delete commented out code?

**Review by mustafasoylu — CHANGES_REQUESTED** [review]

File: src/somesy/core/writer.py (hunk 2 above)
Comment (apirogov) [reply src/somesy/core/writer.py:267]: that's actually a point maybe needs a bit of discussion. 

In `pom.xml` there's developers and contributors, and contributors is apparently for minor stuff (compared to developers), from the POM reference:

```
Contributors are like developers yet play an ancillary role in a project's lifecycle. Perhaps the contributor sent in a bug fix, or added some important documentation. A healthy open source project will likely have more contributors than developers.
```

In `package.json` you apparently dump **all** people, and just a single author is highlighted. So I'm not fully sure whether and how these semantics can be made to match and how best represent it in code.

If we could harmonize it, maybe having the contributors handled in the core/writer class would make sense... but I guess for now I'll remove the commented part.

**Review by mustafasoylu — APPROVED** [review]
LGTM

**Pull request merged**